### ISSUE TITLE
How to properly close and restart server?

### ISSUE THREAD

**Issue by watho** [felixblaschke/shelf_plus#33]
Hi, 

thanks for this project. I just started to learn flutter and it helps me a lot to implement a simple server. However i have a question which might be related to my limited understanding.

I need to stop and restart the server multiple times. Especially for testing purposes. The following simple test case does not work for me.

```dart
import 'package:flutter_test/flutter_test.dart';
import 'package:shelf_plus/shelf_plus.dart' as shelf_plus;

void main() {
  group('Test Shelf Server', () {
    test('Test start and stop', () async {
      var context1 = await shelf_plus.shelfRun(
        init,
        defaultBindAddress: '0.0.0.0',
        defaultBindPort: 8085,
        defaultEnableHotReload: false,
      );
      await context1.close().whenComplete(() => print("context 1 closed"));
      //await Future.delayed(const Duration(seconds: 5));
      var context2 = await shelf_plus.shelfRun(
        init,
        defaultBindAddress: '0.0.0.0',
        defaultBindPort: 8085,
        defaultEnableHotReload: false,
      );
      await context2.close().whenComplete(() => print("context 2 closed"));
    });
  });
}

shelf_plus.Handler init() {
  var app = shelf_plus.Router().plus;
  app.get('/<ignored|.*>', (request) => 'Congratulation! This works.');
  return app;
}
```

Error Message
---------------------
```
Testing started at 15:17 ...

Shell: shelfRun HTTP service running on port 8085
context 1 closed
package:shelf_plus/src/shelf_run.dart 51:49  shelfRun.<fn>
package:shelf_plus/src/shelf_run.dart 81:20  shelfRun

SocketException: Failed to create server socket (OS Error: The shared flag to bind() needs to be `true` if binding multiple times on the same (address, port) combination.), address = 0.0.0.0, port = 8085
```

I have already looked into the documentation of the BIND-flag but it seems to me this is not what i want. The server should not share the request. It should be only one server available.

**Comment by felixblaschke** [comment]
Hey 😄 thanks for reaching out. No, your code is looking good. I investigated why it wasn't working and found an issue when having set `defaultEnableHotReload` set to `false`.

I published an updated version `1.8.0` of `shelf_plus`. Can you try this out and confirm that it correctly addressed your issue?

Many thanks.

**Comment by watho** [comment]
Thanks for the quick response. My test case works now. I am looking forward to integrate it in my app.
